### PR TITLE
test: pin adf_to_text edge-case behavior

### LINKED ISSUE #203: test: expand adf_to_text edge-case coverage (empty docs, whitespace blockquotes, multi-hardBreak, mark permutations)
## Context

Deferred from PR #202 review (which replaced the hand-rolled `adf_to_text` renderer with a stateful `AdfRenderer` supporting ordered lists, inline marks, blockquotes, tables, etc.). The new test suite covers all 9 acceptance-criteria features behaviorally (55 adf tests total including a rich snapshot and a structural roundtrip), but several defensive edge-case gaps were flagged by the `pr-test-analyzer` and `silent-failure-hunter` review agents. All are low-priority — current behavior is either correct or a graceful fallback — but adding tests documents the behavior and prevents future regressions.

## Gaps

- [ ] **Empty document** — `{"type": "doc", "content": []}` returns empty string. No explicit test pins this behavior.
- [ ] **Whitespace-only / empty-paragraph blockquote** — a blockquote containing only an empty paragraph currently produces no output at all (the prefix loop skips `if !line.is_empty()`). Arguably the blockquote should still show a `> ` marker; at minimum a test should document what actually happens.
- [ ] **Multiple consecutive `hardBreak` nodes** — `text → hardBreak → hardBreak → text` should render two newlines between segments. Not currently asserted.
- [ ] **Mark permutations** — only `[strong, em]` is tested for multi-mark composition. Additional combinations like `[code, strong]`, `[strike, em]`, `[link, strong]` would tighten the contract.

## Non-goals

This issue is not about adding new features or changing current behavior. Only adding tests to document what the renderer currently does.

## Related

- PR #202 — introduced the rewrite
- Issue #197 — the write-path feature that exposed the need for this read-path update

### PR DESCRIPTION
**Pull request by Zious11**
## Summary

Closes #203.

Adds 7 behavioral tests to `src/adf.rs` that pin the current `adf_to_text` behavior for edge cases flagged by the `pr-test-analyzer` and `silent-failure-hunter` review agents during PR #202. Test-only — no production code changes, no behavior changes. The intent is regression prevention, not correctness fixes.

**Gap coverage (from issue body):**
- Empty document → `""`
- Blockquote containing only an empty paragraph → `""` (quirk of current trailing-empty-lines trim)
- Consecutive `hardBreak` nodes produce a blank line between surrounding text
- Multi-mark combinations: `[code, strong]`, `[strike, em]`, `[link, strong]`

**Additional hardening from local review:**
- Trailing consecutive `hardBreak` exercises the `finish()` `trim_end()` path — the "no stray blank line at end of doc" contract is the more brittle complement to the interior-hardBreak case.

## Test Plan

- [x] `cargo test` — 661 → 668 suite total (adf tests 62 → 69)
- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --all-targets -- -D warnings` (zero warnings)
- [x] Local PR review (`pr-test-analyzer` + `code-reviewer`) — addressed all findings; re-review reported clean
- [x] Copilot round 1 — addressed both findings (comment-accuracy fix, duplicate-test removal)
- [x] No behavior changes — all additions are `#[test]` fns in the existing tests module

### CODE REVIEW

File: src/adf.rs [hunk 1]
```diff
@@ -1619,6 +1619,144 @@ mod tests {
...
+            }]
+        });
+        assert_eq!(adf_to_text(&adf), "a\n\nb");
+    }
+
+    #[test]
+    fn test_render_marks_code_and_strong() {
+        // `[code, strong]` is the shape the write-path emits for `**`x`**` —
+        // `push_code` appends `{type: "code"}` after active marks. The `code`
+        // mark is applied innermost regardless of array position, so bold
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/adf.rs]: Fixed in f298113. Rewrote the comment so it accurately describes `[code, strong]` as the reverse-order case being pinned, and notes the write-path's actual `[strong, code]` shape. Comment now matches the `apply_marks` invariant at `src/adf.rs:644-648`.

File: src/adf.rs [hunk 2]
```diff
@@ -1619,6 +1619,144 @@ mod tests {
...
+        let adf = json!({
+            "type": "doc",
+            "content": [{"type": "paragraph", "content": [
+                {"type": "text", "text": "x", "marks": [
+                    {"type": "strong"}, {"type": "code"}
+                ]}
+            ]}]
+        });
+        assert_eq!(adf_to_text(&adf), "**`x`**");
+    }
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/adf.rs]: Fixed in f298113. Dropped `test_render_marks_strong_and_code_reverse_order_equivalent` entirely — you're right that `test_render_strong_with_code_applies_code_innermost` (from PR #202) already pins the same `[strong, code] → **`x`**` contract, so the new test was pure duplication. The `[code, strong]` coverage in `test_render_marks_code_and_strong` (added by this PR) remains as the complementary reverse-order case.

File: src/adf.rs [hunk 3]
```diff
@@ -1619,6 +1619,126 @@ mod tests {
...
     }
 
+    #[test]
+    fn test_adf_to_text_empty_doc() {
+        // `{"type": "doc", "content": []}` has no children — the renderer
+        // iterates an empty array and `finish()` returns an empty string.
+        // Pinned here so a future refactor that starts emitting a placeholder
+        // for empty documents trips a test instead of silently changing output.
+        let adf = json!({"type": "doc", "content": []});
+        assert_eq!(adf_to_text(&adf), "");
```
[the remark this answers is not in the document]
Comment (Zious11) [reply src/adf.rs:1629]: Fixed: updated the PR description to reflect the final count of 7 new tests (from 62 → 69 adf tests; 661 → 668 suite total). The description was written pre-round-1 when the diff still had 8 tests; the duplicate `test_render_marks_strong_and_code_reverse_order_equivalent` was removed in f298113 per your round 1 finding, but I forgot to sync the description. Now consistent with the diff.
